### PR TITLE
[#3] Add ToastExt

### PR DESCRIPTION
**Pull request by lydiasama**
Closed https://github.com/nimblehq/android-common-ktx/issues/3

## What happened 👀

Kotlin provides the ability to extend a class with new functionality without having to inherit from the class or use design patterns such as Decorator. This is done via special declarations called extensions.

## Insight 📝

- [x] Show short toast, long toast, toast with a layout
- [ ] Add Unit Test

### CODE REVIEW

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+import androidx.fragment.app.Fragment
+
+/**
+ * Provide extension functions to Activity/Fragment to capable of showing Toast in a shorter call
+ *
+ * @param message A message to display
+ * @param duration Duration time to display message
+ */
+fun Activity.toastShort(message: CharSequence, duration: Int = Toast.LENGTH_SHORT) {
+    Toast.makeText(this, message, duration).show()
```
Comment (hoangnguyen92dn) [inline common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:19]: In my opinion, by creating the toast like this, we could not control to hide the toast. Should we create a toast instance and manage to hide it as well? 🤔

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ *         rootLayoutRes = R.id.toastRootLayout,
+ *         textViewId = R.id.tvErrorMessage
+ *     )
+ *
+ * @param message A message to display
+ * @param layoutRes Layout resource id of the layout
+ * @param rootLayoutRes Id of the root ViewGroup
+ * @param textViewId Id of TextView that will be set message
+ */
+fun Activity.toastWithLayout(
```
Comment (hoangnguyen92dn) [inline common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:50]: We can't apply this extension for Fragment? 🤔

Comment (lydiasama) [reply common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:50]: I'm afraid not. 🤔

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt (hunk 1 above)
Comment (lydiasama) [reply common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:19]: @hoangnguyen92dn  Can you explain more use cases? I didn't face that issue before.

File: common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt (hunk 2 above)
Comment (doannimble) [reply common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:50]: @lydiasama @hoangnguyen92dn We can do the `Toast` even with `context` afaik 😄

Comment (hoangnguyen92dn) [reply common-ktx/src/main/java/co/nimblehq/common/extensions/ToastExt.kt:50]: @doannimble I think we could get rid of custom view for Toast. It's deprecated on `setView` method.

> Custom toast views are deprecated. Apps can create a standard text toast with the makeText(Context, CharSequence, int) method, or use a Snackbar when in the foreground. Starting from Android Build.VERSION_CODES.R, apps targeting API level Build.VERSION_CODES.R or higher that are in the background will not have custom toast views displayed.

**Review by hoangnguyen92dn — APPROVED** [review]

**Review by doannimble — APPROVED** [review]
LGTM 🚀

**Pull request merged**